shell/edgex-publish-snap.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# Ensure we fail the job if any steps fail
+set -e -o pipefail
+snapcraft login --with $HOME/EdgeX
+
+# Build snap and grab the generated snap name
+SNAP_TO_PUSH=$(snapcraft | awk '/Snapped/ {print $2}')
```

Comment:
This line seems a little iffy to me, while the phrase `Snapped` isn't likely to show up in the build output there is a ton of stuff printed out during the build. I've found while automating to just grab the name of the `*.snap` file to be easier like this:
```
snapcraft push edgexfoundry*.snap
```
The only problem you might run into with this is if you end up building multiple architectures in the same workspace, but then you can filter with the architecture:
```
snapcraft push edgexfoundry*_amd64.snap
```

If you really want to keep with the current method I'd suggest making the string matching more precise a la something like
```
SNAP_TO_PUSH=$(snapcraft | grep -oP 'Snapped \K(edgexfoundry_.+\.snap)')
```

(and you could even add in the architecture if you really wanted to be careful)

Reply:
Ok I think we will be good with `snapcraft push edgexfoundry*.snap` as we will not be building in the same workspace (dynamic build machines)